File: squares-of-a-sorted-array/squares-of-a-sorted-array.java
class Solution {
    public int[] sortedSquares(int[] nums) {
        int n = nums.length;
        int left = 0;
        int right = n - 1;
        int[] new_nums = new int[n];
        
        for (int i = n - 1; i >= 0; i--){
            if(Math.abs(nums[left]) < Math.abs(nums[right])){
                new_nums[i] = nums[right];
                right --;
            }else{
                new_nums[i] = nums[left];
                left ++;
            }
            
            new_nums[i] = new_nums[i] * new_nums[i];
             
        }
               
        return new_nums;
        
        
        
    }
}

File: squares-of-a-sorted-array/squares-of-a-sorted-array.py
class Solution:
    def sortedSquares(self, nums: List[int]) -> List[int]:
        ptr1 = 0
        ptr2 = len(nums) - 1
        ptr3 = len(nums) - 1
        lst = [0] * len(nums)
        
        while ptr3 >= 0:
            if abs(nums[ptr1]) > abs(nums[ptr2]):
                lst[ptr3] = nums[ptr1] * nums[ptr1]
                ptr1 += 1
                ptr3 -= 1
            else:
                lst[ptr3] = nums[ptr2] * nums[ptr2]
                ptr2 -= 1
                ptr3 -= 1
                
        return lst
                
                
                
                
            
        